Realm/Realm.Unity/Editor/Common/RealmWeaver.cs

```diff
@@ -1504,6 +1506,31 @@ private TypeDefinition WeaveRealmObjectHelper(TypeDefinition realmObjectType, Me
             return helperType;
         }
 
+        private bool IsUsingSync()
+        {
+            try
+            {
+                return SearchMethodOccurrence(_references.SyncConfiguration_Constructor);
+            }
+            catch (ArgumentNullException)
```

Comment:
I thought that being explicit could have helped the reader. I checked all the method calls that SearchMethodOccurrence uses and I can't see any other type of Exception beside ArgumentNullException. That's why I specified it that way.
I guess your point is that "it's just safer to use a wider net to catch them all" in case, e.g. the SearchMethodOccurrence is later modified and new exceptions may arise.
I'll update.